### DISCUSSION TITLE
Opened hatches/doors do not cast shadow when closed

### DISCUSSION THREAD

**Discussion by SomeRandomNoobKekeke** [FakeFishGames/Barotrauma#13811] [Bug reports]
### Disclaimers  

- [X] I have searched the issue tracker to check if the issue has already been reported.
- [ ] My issue happened while using mods.

### What happened?

If sub was saved with an opened hatch/door all nearby shadow casting lights will consider it open forever and it won't cast shadow
Yes, it can happen between rounds
Probably related https://github.com/FakeFishGames/Barotrauma/issues/12087 
It probably won't hurt performance to make it "unhidden" once per round

https://github.com/FakeFishGames/Barotrauma/assets/122838333/a23957d1-bd2b-4ed5-ab10-26586f7ae5e3

https://github.com/FakeFishGames/Barotrauma/assets/122838333/99da28a4-8e2a-40e4-882d-7ecbf0c60f4d

### Reproduction steps

Go to sub editor
Open Typhon
Open that hatch
Turn batteries on
Test

### Bug prevalence

Happens every time I play

### Single player or multiplayer?

Single player

### Version

v1.3.0.4

### Which operating system did you encounter this bug on?

Windows

**Comment by SomeRandomNoobKekeke** [comment]
Perfect timing, now it's in the bug report limbo

![](https://cdn.discordapp.com/emojis/1162762103262822532.gif)

**Comment by NotWendy** [comment]
There's actually an issue ticket for this, which got looked at a year ago and was seemingly too hard to address. But hey, you can read Regalis' message about what exactly causes this.
https://github.com/FakeFishGames/Barotrauma/issues/12087

**Reply by SomeRandomNoobKekeke** [reply]
It's a bit different. When he opens first hatch shadow casting works at first and then breaks, while here it's broken from round start

**Comment by NilanthAnimosus** [comment]
Closing as fixed, tested and merged into the codebase for the next patch/unstable.
